web/src/engine/websites/Kakaopage.ts

```diff
@@ -0,0 +1,220 @@
+            const data = await this.getChaptersFromPage(manga, nextCursor);
+            const chapters = data.contentHomeProductList.edges.map(chapter => new Chapter(this, manga, chapter.node.single.productId.toString(), chapter.node.single.title.replace(manga.Title, '').trim()));
+            nextCursor = data.contentHomeProductList.pageInfo.hasNextPage ? data.contentHomeProductList.pageInfo.endCursor : null;
+            chapterList.push(...chapters);
+            run = nextCursor != null;
+        }
+        return chapterList;
+    }
+
+    async getChaptersFromPage(manga: Manga, nextCursor: string): Promise<APIChapters> {
+        const gql = `
+            query contentHomeProductList(
+              $after: String
+              $before: String
+              $first: Int
+              $last: Int
+              $seriesId: Long!
+              $boughtOnly: Boolean
+              $sortType: String
+            ) {
+              contentHomeProductList(
+                seriesId: $seriesId
+                after: $after
+                before: $before
+                first: $first
+                last: $last
+                boughtOnly: $boughtOnly
+                sortType: $sortType
+              ) {
+                totalCount
+                pageInfo {
+                  hasNextPage
+                  endCursor
+                  hasPreviousPage
+                  startCursor
+                }
+                edges {
+                  cursor
+                  node {
+                    ...SingleListViewItem
+                  }
+                }
+              }
+            }
+            fragment SingleListViewItem on SingleListViewItem {
+              id
+              type
+              single {
+                productId
+                id
+                isFree
+                thumbnail
+                title
+                slideType
+              }
+            }
+
+        `;
+        const vars = {
+            seriesId: parseInt(manga.Identifier),
+            boughtOnly: false,
+            sortType: 'asc',
+            after: nextCursor
+        };
+        const request = new FetchRequest(new URL('/graphql', this.URI).href, {
+            headers: {
+                Referer: this.URI.href,
+                Origin: this.URI.href
+            }
+        });
+        const operationName = 'contentHomeProductList';
+        return await FetchGraphQL<APIChapters>(request, operationName, gql, JSON.stringify(vars));
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const gql = `
+            query viewerInfo($seriesId: Long!, $productId: Long!) {
+              viewerInfo(seriesId: $seriesId, productId: $productId) {
+                viewerData {
+                  ...ImageViewerData
+                  __typename
+                }
+                __typename
+              }
+            }
+
+            fragment ImageViewerData on ImageViewerData {
```

Comment:
May be easier to read (and eventually shorter) when just nesting the fragments into a single structure